src/colony/lua_tm.c

```diff
@@ -427,9 +432,13 @@ static int l_tm_ssl_read (lua_State* L)
   uint8_t buf[20000];
   size_t buf_len = sizeof(buf);
   int err = tm_ssl_read(session, buf, &buf_len);
-  (void) err;
+  if (err < 0) {
```

Comment:
I thought all errors were always negative. Oh, I guess you are saying that any != 0 is an error?